### ISSUE TITLE
[feature] export types for `--reporter=json` output

### ISSUE THREAD

**Issue by gameroman** [oxc-project/oxc#21988]
In the `oxlint` and `oxfmt` packages would be nice to have a `JSONReport` type for the `--reporter=json` output

So you could do something like this

```ts
import type { JSONReport } from "oxlint"
```

```ts
import type { JSONReport } from "oxfmt"
```

**Comment by camc314** [comment]
this is a great idea.

For those interested in implementing this feature in oxlint:

The `JSONReport` type should be generated, not handwritten, following the same pattern used for config types like `apps/oxlint/src-js/package/config.generated.ts`.

1. Add JSON Schema generation for the reporter output.
   - For `oxlint`, model the current `--format=json` output from `apps/oxlint/src/output_formatter/json.rs`.
   - The top-level shape is roughly:
     - `diagnostics`
     - `number_of_files`
     - `number_of_rules`
     - `threads_count`
     - `start_time`
   - The diagnostic entries should match what `miette::JSONReportHandler` serializes today.

2. Add debug/test assertions that guard the generated type against drifting from the real JSON output.
   - Define Rust structs for the JSON report shape used by the reporter.
   - Derive `Serialize`, `Deserialize`, and `JsonSchema` from those structs.
   - In tests or `#[cfg(debug_assertions)]` assertions, serialize real reporter output and deserialize it back into the same schema/source struct.
   - For diagnostic entries, make sure the assertion covers output produced by `miette::JSONReportHandler`, since that is the real-world shape users receive.
   - Include at least one fixture with labels/source locations so fields like `span`, `offset`, `line`, and `column` are covered.
   - The goal is that if the actual `--reporter=json` / `--format=json` output changes, generation or tests fail instead of silently publishing stale `JSONReport` types.

3. Generate TypeScript from that schema with `json-schema-to-typescript`, like:
   - `apps/oxlint/scripts/generate-config-types.ts`
   - `apps/oxfmt/scripts/generate-config-types.ts`

4. Add generated files, for example:
   - `apps/oxlint/src-js/package/report.generated.ts`
   - `apps/oxfmt/src-js/report.generated.ts`

5. Re-export the public type from the package entry points:
   - `apps/oxlint/src-js/index.ts`
     ```ts
     export type { JSONReport } from "./package/report.generated.ts";
     ```
   - `apps/oxfmt/src-js/index.ts`
     ```ts
     export type { JSONReport } from "./report.generated";
     ```

6. Add `just` recipes similar to:
   - `just linter-report-ts`

7. Add tests or type checks proving this works:
   ```ts
   import type { JSONReport } from "oxlint";
   ```

Important: don't manually change the generated TypeScript report type. The source of truth should be the Rust/JSON schema output. Changing this file will result in CI failing as we re-generate these in CI and check for a diff
